test/js/bun/util/readablestreamtoarraybuffer.test.ts

```diff
@@ -0,0 +1,29 @@
+import { test, expect } from "bun:test";
+
+test("readableStreamToArrayBuffer works", async () => {
+  // the test calls InternalPromise.then. this test ensures that such function is not user-overridable.
+  let _then = Promise.prototype.then;
+  let counter = 0;
+  // @ts-ignore
+  Promise.prototype.then = (...args) => {
+    counter++;
+    return _then.apply(this, args);
+  };
+  try {
+    const result = await Bun.readableStreamToArrayBuffer(
+      new ReadableStream({
+        async start(controller) {
```

Comment:
nitpick: `start` is executed immediately ,synchronously so this test does not cover promises

it would need to go under pull